File: Leetcode/singleNumber.java
//package Leetcode;
import java.util.HashSet;
import java.util.Set;
public class singleNumber {
    public int singleNumber(int[] nums) {
        Set<Integer> s = new HashSet<>();
        for(int i = 0; i < nums.length; i++){
            int val = nums[i];
            if(!s.contains(val)){
                s.add(val);
            } else {
                s.remove(val);
            }
        }
        int ans = s.iterator().next();
        return ans;
    }
}

File: Vectors/SingleNumber.cpp
#include <iostream>
#include <vector>
using namespace std;

class SingleNumber {
public:
    int singleNumber(vector<int>& nums) {
        int ans = 0;
        for (int val : nums) {
            ans ^= val;
        }
        return ans;
    }
};

int main() {
    SingleNumber s;
    vector<int> nums = {8, 9, 3, 3, 5, 8, 9};
    cout << s.singleNumber(nums) << endl;
    return 0;
}